doc/proxy-metrics.md

```diff
@@ -133,3 +135,63 @@ request_total{
 
 [prom-format]: https://prometheus.io/docs/instrumenting/exposition_formats/#format-version-0.0.4
 [pod-template-hash]: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template-hash-label
+
+# Transport-Level Metrics
+
+The following metrics are collected at the level of the underlying transport
+layer.
+
+### `connection_opens`
+
+A counter of the total number of transport connections which have been opened.
+
+### `connection_closes`
```

Comment:
bikeshedding (curious if there are idioms elsewhere already):

Would it be better to make these i.e. `connection_open_count`, to be explicit that these are an increasing counter?

Reply:
I agree that names should help inform whether things are counters vs. gauges. The Prometheus export format also includes helper comments to aide this as well.

In this particular case, I think `connection_opens` sounds more like a counter than `connection_open_count`, and if this really were a gauge, i'd expect `connections_open`.

Reply:
Sidenote, _do_ we think it's correct for these to be counters? This is (once again) based on what Finagle does, having opens/closes counters _and_ a gauge of currently-open connections.

Reply:
yes, i think they should be counters

Reply:
Okay, that sounds good.